fe/src/main/jflex/sql_scanner.flex

```diff
@@ -439,20 +439,36 @@ QuotedIdentifier = \`(\`\`|[^\`])*\`
 SingleQuoteStringLiteral = \'(\\.|[^\\\'])*\'
 DoubleQuoteStringLiteral = \"(\\.|[^\\\"])*\"
 
-// Both types of plan hints must appear within a single line.
-TraditionalCommentedPlanHints = "/*" [ ]* "+" [^\r\n*]* "*/"
-// Must end with a line terminator.
-EndOfLineCommentedPlanHints = "--" [ ]* "+" {NonTerminator}* {LineTerminator}
-
 FLit1 = [0-9]+ \. [0-9]*
 FLit2 = \. [0-9]+
 FLit3 = [0-9]+
 Exponent = [eE] [+-]? [0-9]+
 DoubleLiteral = ({FLit1}|{FLit2}|{FLit3}) {Exponent}?
 
+EolHintBegin = "--" " "* "+"
+CommentedHintBegin = "/*" " "* "+"
```

Comment:
I think your change will cause problem for query like "select /* test */ a from test"